cloudmarker/events/vmdatadiskencryptionevent.py

```diff
@@ -0,0 +1,122 @@
+"""VM Data disk encryption event.
+
+This module defines the :class:`VMDataDiskEncryptionEvent` class that
+identifies an unencrypted VM data disk. This plugin works on the virtual
+machine properties found in the ``com`` bucket of ``virtual_machine``
+records.
+"""
+
+
+import logging
+
+from cloudmarker import util
+
+_log = logging.getLogger(__name__)
+
+
+class VMDataDiskEncryptionEvent:
+    """VM Data disk encryption event plugin."""
+
+    def __init__(self):
+        """Create an instance of :class:`VMDataDiskEncryptionEvent`."""
+
+    # pylint: disable=R0201
+    def eval(self, record):
+        """Evaluate virtual machine to check for unencrypted data disks.
+
+        Arguments:
+            record (dict): A virtual machine record.
+
+        Yields:
+            dict: An event record representing an unencrypted data disk
+            of a virtual machine
+
+        """
+        # If 'com' bucket is missing, we have a malformed record. Log a
+        # warning and ignore it.
+        com = record.get('com')
+        if com is None:
+            _log.warning('Virtual machine record is missing \'com\' key: %r',
+                         record)
+            return
+
+        # This plugin understands compute rule records only,
+        # so ignore  any other record types.
+        common_record_type = com.get('record_type')
+
+        if common_record_type != 'compute':
+            return
+
+        # If 'ext' bucket is missing, we have a malformed record. Log a
+        # warning and ignore it.
+        ext = record.get('ext')
+        if ext is None:
+            _log.warning('Virtual machine record is missing \'ext\' key: %r',
+                         record)
+            return
+        all_data_disks_encrypted = ext.get('all_data_disks_encrypted')
+
+        if all_data_disks_encrypted:
+            return
+        if com.get('cloud_type') == 'azure':
+            yield from _get_azure_vm_data_disk_encryption_event(
+                com, ext, record.get('raw'))
+
+    def done(self):
+        """Perform cleanup work.
+
+        Currently, this method does nothing. This may change in future.
+        """
+
+
+def _get_azure_vm_data_disk_encryption_event(com, ext, raw):
+    """Evaluate Azure VM for unencrypted data disks.
+
+    Arguments:
+        com (dict): Virtual Machine record `com` bucket
+        ext (dict): Virtual Machine record `ext` bucket
+        raw (dict): Virtual Machine record `raw` bucket
+    Returns:
+        dict: An event record representing unencrypted data disk
+
+    """
+    friendly_cloud_type = util.friendly_string(com.get('cloud_type'))
+    os_disk_name = raw.get('storage_profile').get('os_disk').get('name')
```

Comment:
This needs to become:

```
raw.get('storage_profile', {}).get('os_disk', {}).get('name')
```

so that if `raw.get('storage_profile')` returns `None`, the code still wouldn't crash.